src/main/java/com/direwolf20/buildinggadgets/common/util/helpers/NBTHelper.java

```diff
@@ -345,8 +347,31 @@ public static NBTTagList createStringList(String[] strings) {
     }
 
     /**
-     * If the given stack has a tag, returns it. If the given stack does not have a tag, it will set a reference and return the new tag
-     * compound.
+     * Connect two {@link NBTTagList} together to create a new one. This process has no side effects, which means it
+     * will not modify two parameters.
+     * <p>
+     * Additionally, if any of the lists are empty, the method will directly return the nonempty one. If both of them are
+     * empty, it will directly return the second list.
+     */
+    public static NBTTagList concat(NBTTagList first, NBTTagList second) {
+        if (first.isEmpty())
+            return second;
+        if (second.isEmpty())
+            return first;
+
+        NBTTagList connected = new NBTTagList();
+        for (int i = 0; i < first.size(); i++) {
```

Comment:
Steams would make this a single line if I'm not wrong?

Reply:
There's sadly no Collector for NBTagList, though souch a thing should be easy to write. Providing that as a utility Method might be nice, as all PlacementSequences may be stream based. 
```suggestion
        return Stream.concat(first.stream(), second.stream()).collect(NBTHelper.toTagList());
```

Reply:
Well, that makes me sad :P

Reply:
I just used a few min to write said collector...

Reply:
@hnOsmium0001 please refer to: https://github.com/Direwolf20-MC/BuildingGadgets/commit/b3bf00a6af8a17149d797ef3d3126d1097fd548e